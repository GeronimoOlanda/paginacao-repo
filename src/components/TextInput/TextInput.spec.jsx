import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { TextInput } from '.';

// passando parametros para testes, sao chamados de mocks, parametros usados apenas para testes

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'Funfano'} />);

    const input = screen.getByPlaceholderText(/Digite a sua pesquisa/i);
    expect(input).toBeInTheDocument(); // verifica se o que queremos esta no documento
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn(); // cria uma funcao imaginaria para testes
    render(<TextInput handleChange={fn} searchValue="um valor qualquer" />);

    const input = screen.getByPlaceholderText(/Digite a sua pesquisa/i); //verifica se o que o usuario esta vendo é o que queremos

    const value = 'o valor';

    //userEvent.type recebe dois parametros, o primeiro é a entrada,
    //e depois o valor que ela vai receber para teste
    userEvent.type(input, value);

    expect(input.value).toBe('um valor qualquer');
    expect(fn).toHaveBeenCalledTimes(value.length); // verifica quantas vezes a variavel foi chamada
  });

  it('should math snapshot', () => {
    const fn = jest.fn(); // cria uma funcao imaginaria para testes
    const { container } = render(<TextInput handleChange={fn} searchValue="" />);

    expect(container).toMatchSnapshot();
  });
});
