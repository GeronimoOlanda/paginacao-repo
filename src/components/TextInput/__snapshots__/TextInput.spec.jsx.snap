// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should math snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="Digite a sua pesquisa"
    type="search"
    value=""
  />
</div>
`;
